newsfragments/4410.added.md

```diff
@@ -0,0 +1,5 @@
+* Added bindings for `PyList_GetItemRef` on Python 3.13 and newer. Also added
+  `ffi::compat::PyList_GetItemRef` which re-exports the FFI binding on Python
+  3.13 and newer and defines a compatibility shim on older versions. This
+  function returns an owned reference to the item in the list and should be
+  preferred if the list is not known *a priori* to be shared between threads.
```

Comment:
For what it's worth, I would probably just go for the shorter:

```suggestion
Add ffi binding `PyList_GetItemRef` and `pyo3::compat::PyList_GetItemRef`
```

... as users using the raw C FFI should hopefully be aware of what they need to be doing for free threading support anyway.